prov/sm2/src/sm2.h

```diff
@@ -90,6 +90,11 @@ enum {
 	sm2_proto_return,
 	sm2_proto_cma,
 	sm2_proto_ipc,
+	/* Used by receiver to send the IPC handle to the sender */
+	sm2_proto_ipc_host_to_dev,
+	/* Used by sender to notify the receiver that the device memcpy is
+	   complete */
+	sm2_proto_ipc_host_to_dev_ack,
```

Comment:
I've addressed this by getting rid of the two new protocols and instead using flags in the header. I split the 32 bit proto into 16 bit proto and 16 bit proto_flags. So now instead of setting a different protocol every time, I just change the flags.

This does make the recv path a bit messy but we are planning a refactor where we switch on protocol before switching on op and that would make this a lot cleaner.